art/attacks/evasion/graphite/graphite_whitebox_pytorch.py

```diff
@@ -0,0 +1,565 @@
+        if self.estimator.clip_values is not None:
+            clip_min, clip_max = self.estimator.clip_values
+        else:
+            clip_min, clip_max = np.min(x), np.max(x)
+
+        # Some initial setups
+        x_adv = x.astype(ART_NUMPY_DTYPE)
+
+        if not self.estimator.channels_first:
+            x = np.transpose(x, (0, 3, 1, 2))
+            x_adv = np.transpose(x_adv, (0, 3, 1, 2))
+            mask = np.transpose(mask, (0, 3, 1, 2))
+
+        y_pred = self.estimator.predict(
+            np.transpose(
+                convert_to_network(np.transpose(x_adv[0], (1, 2, 0)), self.net_size, clip_min, clip_max)[
+                    np.newaxis, :, :, :
+                ],
+                (0, 3, 1, 2),
+            )
+        )
+        if is_probability(y_pred):
+            self.use_logits = False
+        else:
+            self.use_logits = True
+
+        # Generate the adversarial samples
+        for i in range(x_adv.shape[0]):
+            x_adv[i] = self._perturb(
+                x=x_adv[i],
+                y=y[i],
+                mask=mask[i],
+                pts=pts,
+                obj_width=obj_width,
+                focal=focal,
+                clip_min=clip_min,
+                clip_max=clip_max,
+            )
+
+        y = to_categorical(y, self.estimator.nb_classes)  # type: ignore
+
+        # COMPUTE SUCCESS RATE
+        x_copy = np.zeros((x.shape[0], self.net_size[1], self.net_size[0], x.shape[1]))
+        x_adv_copy = np.zeros((x_adv.shape[0], self.net_size[1], self.net_size[0], x_adv.shape[1]))
+        x_copy_trans = np.transpose(x, (0, 2, 3, 1))
+        x_adv_copy_trans = np.transpose(x_adv, (0, 2, 3, 1))
+        for i in range(x_copy.shape[0]):
+            x_copy[i] = convert_to_network(x_copy_trans[i], self.net_size, clip_min, clip_max)
+            x_adv_copy[i] = convert_to_network(x_adv_copy_trans[i], self.net_size, clip_min, clip_max)
+
+        if self.estimator.channels_first:
+            x_copy = np.transpose(x_copy, (0, 3, 1, 2))
+            x_adv_copy = np.transpose(x_adv_copy, (0, 3, 1, 2))
+
+        logger.info(
+            "Success rate of GRAPHITE white-box attack: %.2f%%",
+            100
+            * compute_success(
+                self.estimator,
+                x_copy.astype(np.float32),
+                y,
+                x_adv_copy.astype(np.float32),
+                self.targeted,
+                batch_size=self.batch_size,
+            ),
+        )
+
+        if not self.estimator.channels_first:
+            x_adv = np.transpose(x_adv, (0, 2, 3, 1))
+
+        return x_adv
+
+    def _eval(
+        self,
+        x: "torch.Tensor",
+        x_adv: "torch.Tensor",
+        mask: "torch.Tensor",
+        target_label: int,
+        y_onehot: "torch.Tensor",
+        xforms: List[Tuple[float, float, float, int, float, float, float, float, float]],
+        pts: Optional[np.ndarray],
```

Comment:
Should assign it with a default value of None and move to the end of the arg list.